tests/rptest/services/keycloak.py

```diff
@@ -28,24 +40,64 @@
 LAUNCH_JBOSS_IN_BACKGROUND=1 \
 KEYCLOAK_ADMIN={admin} \
 KEYCLOAK_ADMIN_PASSWORD={pw} \
-{kc} start-dev --http-port={port} --hostname={host} --hostname-port={port} --http-enabled=true --proxy=passthrough \
---log="{log_handler}" --log-file="{logfile}" --log-level="{log_level}" &
+{kc} start-dev
 """
 
 OIDC_CONFIG_TMPL = """\
-http://{host}:{port}/realms/{realm}/.well-known/openid-configuration\
+{scheme}://{host}:{port}/realms/{realm}/.well-known/openid-configuration\
 """
 
+DEFAULT_CONFIG = {
+    'http-port': KC_PORT,
+    'hostname': None,
+    'hostname-port': KC_PORT,
+    'http-enabled': True,
+    'proxy': 'passthrough',
+    'log': KC_LOG_HANDLER,
+    'log-file': KC_LOG_FILE,
+    'log-level': KC_ROOT_LOG_LEVEL,
+}
+
+
+class KeycloakConfigWriter:
+    _cfg = DEFAULT_CONFIG.copy()
+
+    def __init__(self, dir: str = KC_CONF_DIR, extra_cfg: dict = {}):
+        self._cfg.update(extra_cfg)
+        self._dir = dir
+        self._fname = os.path.join(self._dir, KC_CFG)
+        self._tmpdir = tempfile.TemporaryDirectory()
+
+    @property
+    def rep(self):
+        return json.dumps(self._cfg)
+
+    def write(self, node):
+        try:
+            node.account.mkdirs(self._dir)
+        except:
+            pass
+        tmp_f = os.path.join(self._tmpdir.name, KC_CFG)
+        with open(tmp_f, 'w') as f:
+            for k, v in self._cfg.items():
+                if v is None:
+                    continue
+                f.write(f"{k}={str(v).lower() if type(v) is bool else v}\n")
+
+        node.account.copy_to(tmp_f, self._fname)
```

Comment:
if you are using the temporary directory with a single file so you can get the data flushed and visible to the copy command, you can still do this with a normal temporary file

```python
with tempfile.NamedTemporaryFile(delete_on_close=False) as f:
    f.write(...)
    f.close() # makes data visible to next line
    node.account.copy_to(f.name, ...)
```

Reply:
nice one, thanks 👍